server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/views/agents/agent_row_widget.js.msx

```diff
@@ -34,15 +34,44 @@ define(['mithril', 'lodash'], function (m, _) {
     return "/go/agents/" + uuid + "/job_run_history";
   };
 
+  var createBuildDetailsDropDown = function (agent, args) {
+    return (
+      <BuildDetailsWidget agent={agent}
+                          dropDownReset={args.dropDownReset}/>
+    );
+  };
+
   var AgentRowWidget = {
+    controller: function (args) {
+      var ctrl = {
```

Comment:
@ketan @iamsumanth 
Controller can also be used as a [class constructor ](http://mithril.js.org/mithril.component.html#controller-as-a-class-constructor) but it's better to return a value as the functions and variables in it are neatly scoped.